nixos/tests/prometheus-exporters.nix

```diff
@@ -1015,6 +1015,25 @@ let
       '';
     };
 
+    smartctl = {
+      exporterConfig = {
+        enable = true;
+        devices = [
+          "/dev/vda"
+        ];
+      };
+      exporterTest = ''
+        wait_for_unit("prometheus-smartctl-exporter.service")
+        wait_for_open_port("9633")
+        wait_until_succeeds(
+          "curl -sSf 'localhost:9633/metrics'"
+        )
+        wait_until_succeeds(
+            'journalctl -eu prometheus-smartctl-exporter.service -o cat | grep "/dev/vda: Unable to detect device type"'
```

Comment:
Isn't there a way to write a more positive test? I will no insist but there are some docs on how to simulate various SMART values for disks: https://blog.christophersmart.com/2019/12/18/kvm-guests-with-emulated-ssd-and-nvme-drives/

Can and probably pass that as additional arguments to QEMU.

Reply:
The end of the blog post shows what I am seeing in QEMU VMs as well:

```
SMART support is:     Unavailable - device lacks SMART capability.
```

And so on a Hetzner VM you can compare CLI and JSON output:

```
[nix-shell:~]# smartctl --xall /dev/sda
smartctl 7.2 2020-12-30 r5155 [x86_64-linux-5.10.79] (local build)
Copyright (C) 2002-20, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               QEMU
Product:              QEMU HARDDISK
Revision:             2.5+
Compliance:           SPC-3
User Capacity:        20,480,786,432 bytes [20.4 GB]
Logical block size:   512 bytes
LU is thin provisioned, LBPRZ=0
Device type:          disk
Local Time is:        Wed Nov 24 17:19:28 2021 CET
SMART support is:     Unavailable - device lacks SMART capability.
Read Cache is:        Enabled
Writeback Cache is:   Enabled

=== START OF READ SMART DATA SECTION ===
Current Drive Temperature:     0 C
Drive Trip Temperature:        0 C

Error Counter logging not supported

Device does not support Self Test logging
Device does not support Background scan results logging
```

```json
[nix-shell:~]# smartctl --json  --xall /dev/sda
{
  "json_format_version": [
    1,
    0
  ],
  "smartctl": {
    "version": [
      7,
      2
    ],
    "svn_revision": "5155",
    "platform_info": "x86_64-linux-5.10.79",
    "build_info": "(local build)",
    "argv": [
      "smartctl",
      "--json",
      "--xall",
      "/dev/sda"
    ],
    "exit_status": 4
  },
  "device": {
    "name": "/dev/sda",
    "info_name": "/dev/sda",
    "type": "scsi",
    "protocol": "SCSI"
  },
  "vendor": "QEMU",
  "product": "QEMU HARDDISK",
  "model_name": "QEMU QEMU HARDDISK",
  "revision": "2.5+",
  "scsi_version": "SPC-3",
  "user_capacity": {
    "blocks": 40001536,
    "bytes": 20480786432
  },
  "logical_block_size": 512,
  "rotation_rate": 0,
  "device_type": {
    "scsi_value": 0,
    "name": "disk"
  },
  "local_time": {
    "time_t": 1637770772,
    "asctime": "Wed Nov 24 17:19:32 2021 CET"
  },
  "temperature": {
    "current": 0,
    "drive_trip": 0
  }
}
```